File: solutions/python/square-root/1/square_root.py
def square_root(number):
    l: int = 1
    r: int = number
    m: int = (l + r) // 2

    while m * m != number:
        if m * m < number:
            l = m
        else:
            r = m + 1
        m = (l + r) // 2

    return m

File: solutions/zig/square-root/1/square_root.zig
pub fn squareRoot(radicand: usize) usize {
    var l: usize = 0;
    var r: usize = radicand;
    var m: usize = l + (r - l) / 2;
    var sq_m: usize = m * m;

    while (sq_m != radicand) : (sq_m = m * m) {
        if (sq_m < radicand) {
            l = m + 1;
        } else {
            r = m;
        }
        m = l + (r - l) / 2;
    }
    return m;
}
